module/mfrc522_module.c

```diff
@@ -45,9 +46,9 @@ static ssize_t mfrc522_write(struct file *file, const char *buffer, size_t len,
 		pr_info("[MFRC522] With extra data: `%*.s`\n", command.data_len,
 			command.data);
 
-	mfrc522_execute(answer, &command);
+	answer_size = mfrc522_execute(answer, &command);
```

Comment:
We need to check the answer size, it could very well be -1 if an error occured. Plus, if the answer size is zero (like in `mem_write`) then we shouldn't write anything using `pr_info`